test/9.custom-request-clients.test.js

```diff
@@ -0,0 +1,119 @@
+/* global describe, it */
+'use strict'
+
+const request = require('supertest')
+const expect = require('chai').expect
+
+let service
+
+describe('fast-proxy custom request clients', () => {
+  it('init & start remote service', async () => {
+    // init remote service
+    service = require('restana')()
+
+    service.get('/service/redirect', (req, res) => {
+      res.setHeader('location', '/service/redirected')
+      res.send(301)
+    })
+    service.get('/service/redirected', (req, res) => {
+      res.setHeader('url', req.url)
+      res.send()
+    })
+
+    await service.start(3000)
+  })
+
+  describe('with default request clients', () => {
+    let gateway, close, proxy, gHttpServer
+
+    it('init proxy', async () => {
+      const fastProxy = require('../index')({
+        base: 'http://127.0.0.1:3000'
+      })
+      close = fastProxy.close
+      proxy = fastProxy.proxy
+
+      expect(proxy instanceof Function).to.equal(true)
+      expect(close instanceof Function).to.equal(true)
+    })
+
+    it('init & start gateway', async () => {
+      // init gateway
+      gateway = require('restana')()
+
+      gateway.all('/service/*', function (req, res) {
+        proxy(req, res, req.url, {
+          base: req.headers.base,
+          queryString: { age: 33 }
+        })
+      })
+
+      gHttpServer = await gateway.start(8080)
+    })
+
+    it('should 301 on redirected url', async () => {
+      await request(gHttpServer)
+        .get('/service/redirect')
+        .expect(301)
+        .then((response) => {
+          expect(response.headers.url).not.to.equal('/service/redirected')
+        })
+    })
```

Comment:
If redirecting isn't switched on you'll just get a 301 status code, and not be on the new URL. (Within a browser, this might mean you are redirected to a page that you don't have access to without a working proxy.)